UI: Lock around removal from dispatch queue

unknownbrackets opened the pull request:
Happened to glance at this code and noticed the lock was missing.  Can't see any good reason not to have it here, although I guess a click during a view destruct ought to be pretty rare... people do double tap by accident sometimes...

-[Unknown]

unknownbrackets committed:
UI: Lock around removal from dispatch queue.

hrydgard commented:
Having coded a lot of Rust recently, it bothers me to see how many little mistakes like this I've made in PPSSPP that are just entirely impossible to do in Rust. This is one of them :)

unknownbrackets commented:
Well, I think I added these functions, actually.  In my defense, it was before we moved to root.cpp so it wasn't as obvious looking at the code.

-[Unknown]

hrydgard commented:
Well, lots of similar mistakes around the code then :)
